prdoc/pr_7423.prdoc

```diff
@@ -0,0 +1,17 @@
+title: Fix issue with InitiateTransfer and UnpaidExecution
+doc:
+- audience: Runtime Dev
+  description: 
+    Fix issue where setting the `remote_fees` field of `InitiateTransfer` to `None` could lead to unintended bypassing of fees in certain conditions.
+    `UnpaidExecution` is now appended **after** origin alteration.
+    If planning to use `UnpaidExecution`, you need to set `preserve_origin = true`.
+
+    The `AllowExplicitUnpaidExecutionFrom` barrier now allows instructions for receiving funds before origin altering instructions before
+    the actual `UnpaidExecution`.
+    It takes a new generic, `Aliasers`, needed for executing `AliasOrigin` to see if the effective origin is allowed to use `UnpaidExecution`.
```

Comment:
afaict this is not mandatory, but nice-to-have enhancement

```suggestion
    It takes a new generic, `Aliasers` which could be used to validate `AliasOrigin` at the barrier level, before it changes the effective origin and checks whether it's allowed to use `UnpaidExecution`.
```

When passing `Aliasers = ()` everything still works correctly, right? Only that "bad" XCMs it will fail in execution rather than on the barrier, right?

Reply:
As it's done right now, it will fail if it sees an `AliasOrigin` instruction. UnpaidExecution is used mostly in system parachains so it's easier to add this `Aliasers` and make sure they work appropriately with `InitiateTransfer`.

Reply:
Changed the default value of `Aliasers` to `Nothing` and added some more docs and tests to make this behavior more evident